File: examples/python/core/invoke/pass-context-between-invocations.py
@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Pass context between invocations."""
    # First invocation creates context
    initial_context = context.invoke(
        function_name="initialize-context",
        payload=event,
        name="initialize",
    )
    
    # Second invocation uses the context
    processed = context.invoke(
        function_name="process-with-context",
        payload={
            "data": event["data"],
            "context": initial_context,
        },
        name="process",
    )
    
    # Third invocation finalizes
    final_result = context.invoke(
        function_name="finalize",
        payload={
            "processed": processed,
            "context": initial_context,
        },
        name="finalize",
    )
    
    return final_result

File: examples/java/core/invoke/pass-context-between-invocations.java
public class PassContextHandler extends DurableHandler<Map<String, Object>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, Object> event, DurableContext ctx) {
        // First invocation creates context
        var initialContext = ctx.invoke(
            "initialize",
            "initialize-context",
            event,
            Map.class
        );

        // Second invocation uses the context
        var processed = ctx.invoke(
            "process",
            "process-with-context",
            Map.of("data", event.get("data"), "context", initialContext),
            Map.class
        );

        // Third invocation finalizes
        var finalResult = ctx.invoke(
            "finalize",
            "finalize",
            Map.of("processed", processed, "context", initialContext),
            Map.class
        );

        return finalResult;
    }
}
